### PR TITLE
Remove timef wrapper and replace it with mpi_wtime function

### LINKED ISSUE #310: Remove timef wrapper and replace with mpi_wtime function
Per Dusan's suggestion:

> In ufs-weather-model we keep empty function in the main program named timef, in order to satisfy link dependency when we use post library:
> 
> https://github.com/NOAA-EMC/NEMS/blob/09aac909390e7bcacd1cba17b289621b69c8e753/src/MAIN_NEMS.F90#L672
> 
> As you can see timef does nothing in these stubs. timef is a non standard Fortran function. We want to remove it from the ufs.
> 
> timef is currently used in post only in PROCESS.f:
> 
> $ grep -i timef *
> CMakeLists.txt:    TIMEF.f
> 
> grib2_module.f:  real(8), EXTERNAL :: timef                   ! <--- not used, should be removed
> grib2_module.f:!    etime=timef()                             ! <--- not used, should be removed
> 
> INITPOST_GFS_SIGIO.f:      real timef                         ! <--- not used
> 
> PROCESS.f:      real(kind=8)       :: timef,btim
> PROCESS.f:      btim = timef()
> PROCESS.f:      ETAFLD2_tim = ETAFLD2_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      ETA2P_tim = ETA2P_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      MDL2SIGMA_tim = MDL2SIGMA_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      MDL2AGL_tim = MDL2AGL_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      SURFCE2_tim = SURFCE2_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      CLDRAD_tim = CLDRAD_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      MISCLN_tim = MISCLN_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      MDL2STD_tim = MDL2STD_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      FIXED_tim =  FIXED_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      MDL2THANDPV_tim = MDL2THANDPV_tim +(timef() - btim)
> PROCESS.f:      btim = timef()
> PROCESS.f:      CALRAD_WCLOUD_tim = CALRAD_WCLOUD_tim +(timef() - btim)
> 
> TIMEF.f:!       Eta for users with etime but not timef functionality (like  certain
> TIMEF.f:!mp     HPs)  Designed to duplicate timef (elapsed time in milliseconds)
> TIMEF.f:        function timef()
> TIMEF.f:        real*8 timef, etime
> TIMEF.f:!        timef=etime(et)
> TIMEF.f:!        timef=timef*1.e3
> TIMEF.f:        timef=mpi_wtime()
> 
> In order to do that we need to either replace timef() with mpi_wtime(), which is basically what timef is defined to do in TIMEF.f.
> 
> Or leave that function but update CMakeList.txt to move TIMEF.f to the library.
> 
> My suggestion is to remove timef and use mpi_wtime directly in PROCESS.f, and clean up the other few files.

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
This PR removes the local function timef (and TIMEF.f file) and replace calls to this function with direct calls to mpi_wtime().

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
The UPP regression tests were conducted on WCOSS-Dell, WCOSS-Cray and Hera. No changed results were found.

**Review by WenMeng-NOAA — APPROVED** [review]

File: sorc/ncep_post.fd/PROCESS.f [hunk 1]
```diff
@@ -43,30 +43,31 @@
...
 !!     LANGUAGE: FORTRAN
 !!     MACHINE : CRAY C-90
 !!
       SUBROUTINE PROCESS(kth,kpv,th,pv,iostatusD3D)
-
 !
 !----------------------------------------------------------------------------
-!     
+!
+      use mpi, only: mpi_wtime
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/PROCESS.f:55]: @DusanJovic-NOAA I saw other UPP routines using 'include "mpif.h" '. Are there any differences of these two statements on runtime performance?

Comment (DusanJovic-NOAA) [reply sorc/ncep_post.fd/PROCESS.f:55]: No, there shouldn't be any difference in runtime performance.

Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/PROCESS.f:55]: Thanks!

**Review by fossell — APPROVED** [review]

**Pull request merged**